package messages;

message WorldModel
{
    optional int64 timestamp = 1;

    optional float my_x = 2;
    optional float my_y = 3;
    optional float my_h = 4;

    optional float my_x_uncert = 5;
    optional float my_y_uncert = 6;
    optional float my_h_uncert = 7;

    optional bool  ball_on = 8;

    optional float ball_dist = 9;
    optional float ball_bearing = 10;

    optional float ball_dist_uncert = 11;
    optional float ball_bearing_uncert = 12;

    optional float chase_time = 13;
    optional float defender_time = 14;
    optional float offender_time = 15;
    optional float middie_time = 16;

    optional int32 role = 17;
    optional bool  in_kicking_state = 18;

    optional bool  active = 19;
}